met-proxy.py

```diff
@@ -338,71 +336,55 @@ async def getHolisticResponse(lat: float, lon: float, userAgent: str, locationIq
-        printColor(
-            f" - Serving Incoming request {bcolors.BOLD}{self.path} with User-Agent: {userAgent}",
-            color=bcolors.PURPLE)
+async def handleRequest(request: web.Request) -> web.Response:
+    userAgent = userAgentDefault
+    if allowOverrideUserAgent:
+        userAgent = request.headers.get("User-Agent", userAgentDefault)
 
-        try:
-            lat, lon, locationIqApiKey = self.parseIncomingRequest()
-            with cacheLock:
-                dataExpires, data = cache.get((lat, lon), (None, None))
-            if data is None or dataExpires < time.time():
-                data = eventLoop.run_until_complete(getHolisticResponse(lat, lon, userAgent, locationIqApiKey))
-            else:
-                printColor(f"Serving cached data for {self.path} - cache expires in {int(dataExpires - time.time())} seconds", color=bcolors.YELLOW)
-
-            self.send_response(200)
-            self.send_header("Content-type", "application/json")
-            self.end_headers()
-            printColor(
-                f" - Serving data for {self.path}", color=bcolors.BOLD + bcolors.GREEN)
-            if debug:
-                pprint(json.loads(data), compact=True)
-                sys.stdout.flush()
-
-            self.wfile.write(data)
-        except BrokenPipeError:
-            printColor(
-                "Broken pipe - won't respond to the client",
-                color=bcolors.RED)
-        except HTTPError as e:
-            printColor(f"HTTP Error requesting {e.url}: {e}", color=bcolors.RED)
-            self.send_error(e.code, e.msg)
-        except BaseException:
-            printColor(traceback.format_exc(), color=bcolors.RED)
-            self.send_error(408)
+    printColor(
+        f" - Serving Incoming request {bcolors.BOLD}{request.path_qs} with User-Agent: {userAgent}",
+        color=bcolors.PURPLE)
+
+    try:
+        lat, lon, locationIqApiKey = parseRequest(request.query)
+
+        with cacheLock:
+            dataExpires, data = cache.get((lat, lon), (None, None))
+
+        if data is None or dataExpires < time.time():
+            data = await getHolisticResponse(lat, lon, userAgent, locationIqApiKey)
+        else:
+            printColor(f"Serving cached data for {request.path_qs} - cache expires in {int(dataExpires - time.time())} seconds", color=bcolors.YELLOW)
+
+        printColor(
+            f" - Serving data for {request.path_qs}", color=bcolors.BOLD + bcolors.GREEN)
+        if debug:
+            pprint(json.loads(data), compact=True)
+            sys.stdout.flush()
+
+        return web.Response(text=data.decode('utf-8'), content_type="application/json")
+    except web.HTTPError as e:
+        print(f"HTTP Error requesting {e}")
+        return e
+    except BaseException:
+        printColor(traceback.format_exc(), color=bcolors.RED)
+        return web.HTTPRequestTimeout(reason="Error: Request Timeout")
```

Comment:
I'm not sure we need any of this either? I just tried to keep the 408 Request Timeout from the old code. If we remove it, any generic exceptions will instead give 500 Internal Server Error.